tinygrad/lazy.py

```diff
@@ -210,6 +210,12 @@ def shuffle_and_prune_movement_ops(self, st: ShapeTracker, op: MovementOps, arg:
   def reduce_op(self:LazyBuffer, op:ReduceOps, new_shape:Tuple[int, ...]) -> LazyBuffer:
     if self.shape == tuple(new_shape): return self
     srcs = _push_movement_ops((self,)) if SHUFFLE_MOVEMENT_OPS else (self,)
+
+    # Split multi-axis reduction into two kernels on Metal for perf
+    if sum(shape_differences := [new != old for new, old in zip(new_shape, self.shape)]) > 1 and self.device == "METAL":
```

Comment:
Assuming we keep the device check, it should probably come first to avoid the sum on other backends.

Fwiw the first iteration of this pr was a lot faster on cuda, though not as fast as pytorch. Haven't tested the current one.